packages/babel-generator/src/printer.ts

```diff
@@ -770,17 +834,31 @@ class Printer {
       return;
     }
 
-    let lines = 0;
+    if (!newLine) {
+      return;
+    }
+
+    const startLine = opts.nextNodeStartLine;
+    const lastCommentLine = this._lastCommentLine;
+    if (startLine > 0 && lastCommentLine > 0) {
+      const offset = startLine - lastCommentLine;
+      this.newline(offset || 1);
+      return;
+    }
+
     // don't add newlines at the beginning of the file
     if (this._buf.hasContent()) {
-      if (!leading) lines++; // always include at least a single line after
-      if (opts.addNewlines) lines += opts.addNewlines(leading, node) || 0;
+      // let lines = 0;
```

Comment:
Also here, why do we keep the old code?

Reply:
This is the original logic of wrapping lines according to the node layout, and I'm a little hesitant to comment or remove them. It might be useful if someone wants to enhance or reference in the future, but not sure if that will happen.🤦‍♂️